cmd/podman/common/build.go

```diff
@@ -239,6 +236,36 @@ func ParseBuildOpts(cmd *cobra.Command, args []string, buildOpts *BuildFlagsWrap
 		default:
 			return nil, fmt.Errorf("no Containerfile or Dockerfile specified or found in context directory, %s: %w", contextDir, syscall.ENOENT)
 		}
+	} else {
```

Comment:
Im struggling a bit with this if clause.  Part of me thinks we should run the iterator without a conditional, meaning if the len is 0, it will do nothing right?  And then go through the len == 0 ?  Another part of me thinks this is getting elongated and should be broken up so we are calling funcs like _areContainerfilesValid_ ... really I could go either way or nothing.  Just raising as a nit,  Code will work just thinking about the next person ...   @Luap99 any comment here?

Reply:
yes please split this into a function. This whole code is already unreadable (not your fault of course) but adding more does not help us either.

I think the point of the else is the the len(containerFiles) == 0 check above we modify the containerFiles slice in there so without the else we might iterate over the new times. I have not looked if this makes a practical difference.